docs/development/localnet-with-kubernetes.md

```diff
@@ -0,0 +1,69 @@
+- [postgres-operator](https://github.com/zalando/postgres-operator) - the operator to manage postgres lifecycle in kubernetes;
+  - once operator is available, `tilt` pushes a [Postgres CRD](../../build/localnet/postgres-database.yaml) to the cluster to provision a postgres database;
+- [obeservability-stack](../../build/localnet/observability-stack/) - a helm chart that installs the observability stack (prometheus, grafana, etc.) that is used to gather metrics and logs from localnet services.
+- v1 cli client - this is a binary that can be used to perform operations on testnet, e.g. you can run `make localnet_client_debug` to execute commands such as `ResetToGenesis`, or `TogglePacemakerMode`. This binary is also automatically updated when you make changes to the codebase.
+
+Tilt continuously monitors files on local filesystem, and when it detects a change, it will rebuild the binary once and distribute it to all the pods that are running. This allows you to iterate on the code and see the changes immediately.
+This works for both - pocket v1 codebase, and pocket-operator codebase that lives in a separate repo/directory.
+
+The workloads however are not managed by Tilt directly, but by the pocket-operator. Tilt instructs pocket-operator by pushing the CRDs to the cluster, and the operator takes care of the rest. This allows us to test and validate pocket-operator, however if we notice this set up introduces too much friction - we will modify the `Tiltfile` to manage the workloads on kubernetes directly.
+
+### Troubleshooting
+
+Sometimes you might experience issues running localnet on Kubernetes. They might be related to the fact different people run different clusters/OS/environments, and sometimes our laptops can go to sleep and that might not play well with virtual machines, postgres or pocket client.
+
+- Check tilt web UI by pressing a space in the terminal where you started tilt or going to [this page](http://localhost:10350/) in your browser. If you see any errors, you can click "Trigger Update" on a resource that has issues to restart the service or retry a command.
+- If triggering an update didn't help, please try to run `make localnet_down` and then `make localnet_up` again. This will clean up most of the resources and start the localnet from scratch.
+- If `make localnet_down` didn't help, we suggest to rebuild local kubernetes cluster using the tool you're managing your cluster with - it could be Docker Desktop, Rancher Desktop, k3s, k3d, minikube, etc.
+- Open an issue in this repo if you're still experiencing issues with running localnet using this guide.
+
+### How to change configuration files
+
+LocalNet resources are managed by pocket-operator, so if you'd like to change a config file (e.g. genesis file), you can do it by modifying the corresponding CRD in `pocket-operator` repo.
```

Comment:
I think I should expand on that, but this should probably go into the pocket-operator documentation and linked here.